tools/integration_tests/tokenserver/test_support.py

```diff
@@ -151,58 +188,74 @@ def _get_node(self, id):
             "backoff": backoff,
         }
 
-    def _last_insert_id(self):
-        cursor = self._execute_sql("SELECT LAST_INSERT_ID()", ())
-        (id,) = cursor.fetchone()
-        cursor.close()
-
-        return id
-
-    def _add_service(self, service_name, pattern):
-        query = "INSERT INTO services (service, pattern) \
-            VALUES(%s, %s)"
-        cursor = self._execute_sql(query, (service_name, pattern))
+    def _add_service(self, service, pattern):
+        """Add definition for a new service."""
+        if self.db_mode == "postgres":
+            insert_sql = sqltext("""
+            insert into services (service, pattern)
+            values (:service, :pattern)
+            RETURNING id
+        """)
+        else:
+            insert_sql = sqltext("""
+          insert into services (service, pattern)
+          values (:service, :pattern)
+        """)
+        cursor = self._execute_sql(insert_sql, {"service": service, "pattern": pattern})
         cursor.close()
-
-        return self._last_insert_id()
+        if self.db_mode == "postgres":
+            return cursor.fetchone()[0]
+        else:
+            return cursor.lastrowid
 
     def _add_user(
         self,
         email=None,
+        nodeid=NODE_ID,
         generation=1234,
+        keys_changed_at=1234,
         client_state="aaaa",
         created_at=None,
-        nodeid=NODE_ID,
-        keys_changed_at=1234,
         replaced_at=None,
     ):
-        query = """
-            INSERT INTO users (service, email, generation, client_state, \
-                created_at, nodeid, keys_changed_at, replaced_at)
-            VALUES (%s, %s, %s, %s, %s, %s, %s, %s);
-        """
+        if self.db_mode == "postgres":
+            insert_sql = sqltext("""\
+            insert into users (service, email, nodeid, generation, keys_changed_at, client_state, created_at, replaced_at)
+            values (:service, :email, :nodeid, :generation, :keys_changed_at, :client_state, :created_at, :replaced_at)
```

Comment:
(very much nit on formatting) I guess we don't have linter for this python code?  These lines are long but also inconsistent with those in the `else` block.